kiwi/package_manager/apt.py

```diff
@@ -0,0 +1,260 @@
+            ] + chroot_apt_get_args + self.custom_args + [
+                'install'
+            ] + self.__package_requests(), self.command_env
+        )
+
+    def process_delete_requests(self, force=False):
+        """
+        Process package delete requests (chroot)
+
+        :param bool force: unused
+        """
+        chroot_apt_get_args = self.root_bind.move_to_root(
+            self.apt_get_args
+        )
+        return Command.call(
+            [
+                'chroot', self.root_dir, 'apt-get'
+            ] + chroot_apt_get_args + self.custom_args + [
+                'remove'
+            ] + self.__package_requests(), self.command_env
+        )
+
+    def update(self):
+        """
+        Process package update requests (chroot)
+        """
+        chroot_apt_get_args = self.root_bind.move_to_root(
+            self.apt_get_args
+        )
+        return Command.call(
+            [
+                'chroot', self.root_dir, 'apt-get'
+            ] + chroot_apt_get_args + self.custom_args + [
+                'upgrade'
+            ],
+            self.command_env
+        )
+
+    def process_only_required(self):
+        """
+        Setup package processing only for required packages
+
+        There is no required/recommends information in the rhel repo data
+        """
+        pass
+
+    def match_package_installed(self, package_name, apt_get_output):
+        """
+        Match expression to indicate a package has been installed
+
+        This match for the package to be installed in the output
+        of the apt-get command is not 100% accurate. There might
+        be false positives due to sub package names starting with
+        the same base package name
+
+        :param list package_list: list of all packages
+        :param string log_line: apt-get status line
+        """
+        return re.match(
+            '.*Unpacking ' + re.escape(package_name) + '.*', apt_get_output
+        )
+
+    def match_package_deleted(self, package_name, apt_get_output):
+        """
+        Match expression to indicate a package has been deleted
+
+        :param list package_list: list of all packages
+        :param string log_line: apt-get status line
+        """
+        return re.match(
+            '.*Removing ' + re.escape(package_name) + '.*', apt_get_output
+        )
+
+    def database_consistent(self):
+        """
+        Check if package database is consistent
+
+        There is no database consistency/rebuild for apt-get
+        """
+        pass
+
+    def dump_reload_package_database(self, version=45):
+        """
+        Dump and reload the package database to match the desired db version
+
+        There is no such reload cycle for apt-get
+
+        :param string version: unused
+        """
+        pass
+
+    def __package_requests(self):
+        items = []
+        items += self.package_requests
+        self.cleanup_requests()
+        return items
```

Comment:
You want to make a copy here, so the `package_requests` modification won't affect `items`? Then:
```python
items = self.package_requests[:]
self.cleanup_requests()
return items
```